digdag-core/src/main/java/io/digdag/core/repository/ProjectControl.java

```diff
@@ -98,22 +100,30 @@ private void updateSchedules(
         throws ResourceConflictException
     {
         ImmutableList.Builder<Schedule> schedules = ImmutableList.builder();
+        Map<Long, Scheduler> wfIdToSchedulers = new HashMap<>();
         for (StoredWorkflowDefinition def : defs) {
             Optional<Scheduler> sr = srm.tryGetScheduler(revision, def);
             if (sr.isPresent()) {
                 ScheduleTime firstTime = sr.get().getFirstScheduleTime(currentTime);
                 Schedule schedule = Schedule.of(def.getName(), def.getId(), firstTime.getRunTime(), firstTime.getTime());
                 schedules.add(schedule);
+                wfIdToSchedulers.put(schedule.getWorkflowDefinitionId(), sr.get());
             }
         }
 
         // TODO validate workflows and sessions
         //   * compile workflow
         //   * validate SubtaskMatchPattern
 
-        // DatabaseProjectStoreManager.updateSchedules has more logic
-        // to match names with existent schedules
-        store.updateSchedules(project.getId(), schedules.build());
+        store.updateSchedules(project.getId(), schedules.build(), (oldStatus, newSched) -> {
```

Comment:
I also considered it but I found that the code becomes confusing...`ProjectControlStore.updateSchedules` wants to get list of `Schedule` and callback (ScheduleUpdateAction) wants to use `Scheduler`. To make it possible, here needs a class that extends Schedule and has a Scheduler...